add contrast test with both black and white text on same surface.

ankur2136 opened the pull request:
#109465 Improve flutter contrast test coverage

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

ankur2136 committed:
add contrast test with both black and white text on same surface.

ankur2136 committed:
add contrast test with both black and white text on same surface.

engine-flutter-autoroll committed:
Roll Flutter Engine from c090055485cf to 50cfc83e2c9b (7 revisions) (#109814)

xu-baolin committed:
[Dismissible]Fix a state lose issue (#108971)

engine-flutter-autoroll committed:
Roll Flutter Engine from 50cfc83e2c9b to 5c30ead197b9 (6 revisions) (#109826)

tgucio committed:
Replace empty Container with const SizedBox (#108672)

ankur2136 committed:
Update test.

ankur2136 committed:
Update test.

engine-flutter-autoroll committed:
Roll Flutter Engine from 5c30ead197b9 to 29fe33348b9e (12 revisions) (#109860)

engine-flutter-autoroll committed:
Roll Flutter Engine from 29fe33348b9e to 7db434b2087d (1 revision) (#109862)

eyebrowsoffire committed:
Generate syntax for plugin registration that works both with and without null safety. (#109480)

ankur2136 committed:
Update test with explicit white background.

ankur2136 committed:
Update test with explicit white background.

Piinks committed:
Update Ballistic animation & ClampingScrollSimulation (#107735)

GaryQian committed:
Update ci.yaml tests to Android API 33 (#109820)

engine-flutter-autoroll committed:
Roll Plugins from 4395e3195be7 to cd586d7d6a40 (5 revisions) (#109866)

Commit:
Temporarily remove a bogus warning until fixed by dwds update (#109793)

* Temporarily remove a bogus warning until fixed by dwds update

* Addressed CR feedback

Commit:
Remove passing of --debugger-module-names flag to frontend server (#109791)

The flag is true by default so the behavior does not change.

Next steps:

Use the flag for updated debugger module names:
  - Frontend server: make the current behavior controlled
    by the flag non-conditional
  - Frontend server: add more debugging names changes under
    the same flag, false by default
  - Dwds: make changes required for the new module names.
  - Flutter tools: when matching dwds changes roll to flutter,
    pass the flag to the frontend server again.

- Cleanup:
  - Frontend server: make new behavior default
  - Flutter tools: remove uses of the flag.
  - Frontend server: remove the flag.

Towards: https://github.com/dart-lang/webdev/issues/1692
Helps: https://github.com/flutter/flutter/issues/106727

engine-flutter-autoroll committed:
Roll Flutter Engine from 7db434b2087d to 544c34102586 (1 revision) (#109870)

gspencergoog committed:
Re-Land "Refactor StrokeAlign to allow double values." (#109591) (#109805)

engine-flutter-autoroll committed:
Roll Flutter Engine from 544c34102586 to 92bc431e097d (3 revisions) (#109873)

thkim1011 committed:
[gen_l10n] Add option to format generated localizations files (#109171)

* init

* fix

* fix 2

* fix 3

* tests

* fix tests

* clarify help text

* fix all tests

* fix formatting?

* add second test

* unused import

* remove print

* trailing spaces

* artifacts is never null

* fix

Jasguerrero committed:
add GeneralInfo project validator to analyze --suggestions (#109874)

engine-flutter-autoroll committed:
Roll Flutter Engine from 92bc431e097d to c1ce554ac5bd (1 revision) (#109876)

jonahwilliams commented on packages/flutter_test/test/accessibility_test.dart:
```diff
@@ -43,6 +43,38 @@ void main() {
+                  style: TextStyle(fontSize: 14.0, color: Colors.black),
+                ),
+                Text(
+                  'this is a black text test2',
```
Yeah, this is the problem I was mentioning - we need to figure out why this is not tripping up the guideline test

keyonghan committed:
mark flaky (#109890)

dependabot[bot] committed:
Bump github/codeql-action from 2.1.18 to 2.1.19 (#109888)

hellohuanlin committed:
[platform_view]fix typo in xcuitest (#109884)

goderbauer committed:
Add some logs to catch flake (#109887)

engine-flutter-autoroll committed:
Roll Flutter Engine from c1ce554ac5bd to a418ffad89a9 (4 revisions) (#109882)

engine-flutter-autoroll committed:
Roll Flutter Engine from a418ffad89a9 to 94d07855d7a4 (3 revisions) (#109893)

engine-flutter-autoroll committed:
Roll Flutter Engine from 94d07855d7a4 to a7868e9567fc (3 revisions) (#109895)

engine-flutter-autoroll committed:
Roll Flutter Engine from a7868e9567fc to a0dcde4fb611 (1 revision) (#109897)

Commit:
[framework] remove restriction on texture layer rasterization (#109900)

engine-flutter-autoroll committed:
Roll Flutter Engine from a0dcde4fb611 to 7fb831136168 (2 revisions) (#109901)

engine-flutter-autoroll committed:
Roll Flutter Engine from 7fb831136168 to 98c567732402 (1 revision) (#109903)

engine-flutter-autoroll committed:
Roll Flutter Engine from 98c567732402 to 835ba6b38df1 (1 revision) (#109909)

engine-flutter-autoroll committed:
Roll Flutter Engine from 835ba6b38df1 to 1e63130d7b4e (1 revision) (#109913)

engine-flutter-autoroll committed:
Roll Flutter Engine from 1e63130d7b4e to 06900c486cee (1 revision) (#109915)

engine-flutter-autoroll committed:
Roll Flutter Engine from 06900c486cee to 0005149dca9b (1 revision) (#109917)

engine-flutter-autoroll committed:
Roll Flutter Engine from 0005149dca9b to d89aae21e9fe (2 revisions) (#109922)

engine-flutter-autoroll committed:
Roll Flutter Engine from d89aae21e9fe to 00a48acb208d (1 revision) (#109928)

engine-flutter-autoroll committed:
Roll Flutter Engine from 00a48acb208d to 94c8b3ef5f38 (2 revisions) (#109931)

engine-flutter-autoroll committed:
Roll Flutter Engine from 94c8b3ef5f38 to e87ddd9819f4 (1 revision) (#109934)

engine-flutter-autoroll committed:
Roll Flutter Engine from e87ddd9819f4 to 9d831cd8ed59 (1 revision) (#109938)

engine-flutter-autoroll committed:
Roll Flutter Engine from 9d831cd8ed59 to 3b0c87fa9f28 (2 revisions) (#109941)

engine-flutter-autoroll committed:
Roll Flutter Engine from 3b0c87fa9f28 to d030ae1a5a67 (1 revision) (#109944)

gspencergoog committed:
Revert "Re-Land "Refactor StrokeAlign to allow double values." (#109805)" (#109955)

This reverts commit b3aba4d9c868aa5a5f8c3a518960f454eca2ba40 because it breaks Google tests.

zanderso committed:
[flutter_tools] Enable custom devices on all channels (#109953)

engine-flutter-autoroll committed:
Roll Flutter Engine from d030ae1a5a67 to bd89852694fe (1 revision) (#109961)

engine-flutter-autoroll committed:
Roll Flutter Engine from bd89852694fe to ebee9c3fb865 (3 revisions) (#109967)

engine-flutter-autoroll committed:
Roll Flutter Engine from ebee9c3fb865 to d311584c791d (1 revision) (#109972)

engine-flutter-autoroll committed:
Roll Flutter Engine from d311584c791d to 7295b1ee8765 (1 revision) (#109982)

engine-flutter-autoroll committed:
Roll Flutter Engine from 7295b1ee8765 to 482f967fa6b2 (1 revision) (#109987)

engine-flutter-autoroll committed:
Roll Flutter Engine from 482f967fa6b2 to b9d670aaddcf (1 revision) (#109990)

engine-flutter-autoroll committed:
Roll Flutter Engine from b9d670aaddcf to 5f732d4e4e7a (1 revision) (#109992)

engine-flutter-autoroll committed:
Roll Flutter Engine from 5f732d4e4e7a to 00bb6056cb80 (1 revision) (#109995)

engine-flutter-autoroll committed:
Roll Flutter Engine from 00bb6056cb80 to e70ba6a3a32d (1 revision) (#110003)

bryanoltman committed:
Add clipBehavior to Card+InkWell example (#109872)

* Add clipBehavior to Card+InkWell example

* Add test for example

* Update comment

engine-flutter-autoroll committed:
Roll Plugins from cd586d7d6a40 to 89fe2a2c0b09 (4 revisions) (#110008)

engine-flutter-autoroll committed:
Roll Flutter Engine from e70ba6a3a32d to 0d182ddb0d1d (1 revision) (#110009)

hannah-hyj committed:
InputDecorator iconColor/prefixIconColor/suffixIconColor (#109988)

engine-flutter-autoroll committed:
Roll Flutter Engine from 0d182ddb0d1d to 60707621bf9b (1 revision) (#110011)

gspencergoog committed:
Remove the FocusScopeNode in the navigator (#109702)

engine-flutter-autoroll committed:
Roll Flutter Engine from 60707621bf9b to ca7acb34c5bc (2 revisions) (#110015)

engine-flutter-autoroll committed:
Roll Flutter Engine from ca7acb34c5bc to 38d2213d99ac (1 revision) (#110026)

engine-flutter-autoroll committed:
Roll Flutter Engine from 38d2213d99ac to 9513c34aabcd (2 revisions) (#110033)

Piinks committed:
Deprecate 2018 text theme parameters (#109817)

engine-flutter-autoroll committed:
0fd04455f deprecate pushPhysicalLayer (flutter/engine#35566) (#110036)

dependabot[bot] committed:
Bump github/codeql-action from 2.1.19 to 2.1.20 (#110044)

engine-flutter-autoroll committed:
Roll Flutter Engine from 0fd04455fd77 to 7252c6406b2c (3 revisions) (#110042)

zanderso committed:
Remove most benchmarks of SkSL warmup (#110040)

QuncCccccc committed:
Fixed leading button size on app bar (#110043)

ricardoamador committed:
Moving ios benchmark tests to prod. (#110049)

* Moving io benchmark tests to prod.

* New tasks must be marked bringup: true

hellohuanlin committed:
[platform_view]retry launch if it fails to launch for xcuitest (#110030)

engine-flutter-autoroll committed:
Roll Flutter Engine from 7252c6406b2c to 258c580a83c6 (3 revisions) (#110051)

engine-flutter-autoroll committed:
Roll Flutter Engine from 258c580a83c6 to ca48808d06f3 (1 revision) (#110054)

engine-flutter-autoroll committed:
Roll Flutter Engine from ca48808d06f3 to 8c29591b89a3 (10 revisions) (#110078)

engine-flutter-autoroll committed:
Roll Flutter Engine from 8c29591b89a3 to c9d0a012817d (1 revision) (#110087)

engine-flutter-autoroll committed:
Roll Plugins from 89fe2a2c0b09 to 3a5e6f3acdd3 (4 revisions) (#110092)

keyonghan committed:
Copy artifacts to `applicationBinaryPath` when specified for build+test separation (#109879)

yaakovschectman committed:
Document tristate value (#110106)

Documentation for Checkbox.value previously did not cover the tristate case.

QuncCccccc committed:
Revert "Fixed leading button size on app bar (#110043)" (#110103)

This reverts commit 7e33cd5ef30b9960d5c8d6fc75b42dabb7ec1f2f.

keyonghan committed:
Update open_jdk version `version:11` (#110110)

QuncCccccc committed:
Revert "Reland: Set IconButton.visualDensity default to VisualDensity.standard (#109432)" (#110119)

This reverts commit 79013b2be8d5e726bb4447bce665f63b5eb9308a.

engine-flutter-autoroll committed:
Roll Flutter Engine from c9d0a012817d to 8cc02dc73e95 (11 revisions) (#110121)

* 94f97c337 Roll Skia from b91dbe274717 to 94602de608e0 (4 revisions) (flutter/engine#35634)

* 6ec08c50e Roll Skia from 94602de608e0 to 8c32b0148770 (7 revisions) (flutter/engine#35636)

* 9195b7a2f Roll Dart SDK from 03506589f2d0 to c064b9bacc4d (1 revision) (flutter/engine#35637)

* f7ec70782 Roll Skia from 8c32b0148770 to a9a05d0bf0c5 (3 revisions) (flutter/engine#35639)

* 15cf53acc Image shader dispose (flutter/engine#35640)

* ee3cd0d8a Roll Skia from a9a05d0bf0c5 to 147232a3cebc (2 revisions) (flutter/engine#35641)

* 5f53ee3b2 [Impeller] Use linear sampling for reads in the texture atlas (flutter/engine#35618)

* 4ff7f7b9a Roll Fuchsia Mac SDK from JocShsJvXYeSVjgvW... to Es8SXkqDOYZ5dgYTq... (flutter/engine#35642)

* 565cc861e Roll Skia from 147232a3cebc to cc63d5dd13b6 (1 revision) (flutter/engine#35643)

* b4af69f02 Roll Dart SDK from c064b9bacc4d to 3042d6196824 (1 revision) (flutter/engine#35644)

* 8cc02dc73 Roll Skia from cc63d5dd13b6 to a4744aa054cf (2 revisions) (flutter/engine#35645)

engine-flutter-autoroll committed:
Roll Flutter Engine from 8cc02dc73e95 to 9dee4b9e4b8f (1 revision) (#110122)

engine-flutter-autoroll committed:
Roll Flutter Engine from 9dee4b9e4b8f to 2ecccf195ec7 (2 revisions) (#110128)

* 68beed75a Refactor - Create `Renderer` interface for CanvasKit/HTML switching (flutter/engine#35502)

* 2ecccf195 [Impeller] Enforce a minimum stroke width (flutter/engine#35576)

engine-flutter-autoroll committed:
Roll Flutter Engine from 2ecccf195ec7 to de4ef8c6f379 (3 revisions) (#110132)

* d1ac8d84e Move extension to TypefaceFontProvider instead of SkFontMgr (flutter/engine#35381)

* 55b46a70a Roll Skia from fad2781bd328 to 6cec033662c7 (3 revisions) (flutter/engine#35649)

* de4ef8c6f Roll Dart SDK from 3042d6196824 to 9be467274ae4 (1 revision) (flutter/engine#35650)

engine-flutter-autoroll committed:
7d0afb03e Roll Clang Windows from 60d276923902 to 41d4067d0b7c (flutter/engine#35655) (#110135)

engine-flutter-autoroll committed:
Roll Flutter Engine from 7d0afb03e363 to 59a61b2300ea (2 revisions) (#110137)

engine-flutter-autoroll committed:
Roll Flutter Engine from 59a61b2300ea to 55c461f28697 (1 revision) (#110145)

engine-flutter-autoroll committed:
Roll Flutter Engine from 55c461f28697 to c17a9fff2e84 (1 revision) (#110148)

engine-flutter-autoroll committed:
Roll Flutter Engine from c17a9fff2e84 to 59fe006561db (1 revision) (#110152)

engine-flutter-autoroll committed:
Roll Flutter Engine from 59fe006561db to 1f929978611e (1 revision) (#110154)

engine-flutter-autoroll committed:
Roll Flutter Engine from 1f929978611e to 24e4568bf477 (1 revision) (#110156)

engine-flutter-autoroll committed:
Roll Flutter Engine from 24e4568bf477 to 0834a5f96a08 (1 revision) (#110160)

engine-flutter-autoroll committed:
Roll Flutter Engine from 0834a5f96a08 to 5d7c74d16242 (2 revisions) (#110166)

eggfly committed:
Fix some typos (#110077)

engine-flutter-autoroll committed:
Roll Flutter Engine from 5d7c74d16242 to d73878d9b8ee (1 revision) (#110171)

engine-flutter-autoroll committed:
Roll Flutter Engine from d73878d9b8ee to 9c16a5c94668 (2 revisions) (#110175)

TahaTesser committed:
Fix `floatingActionButtonAnimator` not being updated (#110067)

engine-flutter-autoroll committed:
Roll Plugins from 3a5e6f3acdd3 to e4cabdf56d0d (3 revisions) (#110179)

guidezpl committed:
Add covariant to `ThemeExtension.lerp` (#105207)

* Add covariant to `ThemeExtension.lerp`

* x

* x

guidezpl committed:
Add option to specify class name (#109865)

fluttergithubbot committed:
Marks Mac_arm64_ios build_ios_framework_module_test to be unflaky (#110183)

engine-flutter-autoroll committed:
Roll Flutter Engine from 9c16a5c94668 to 5ce72b373e1a (1 revision) (#110182)

fluttergithubbot committed:
Marks Mac_ios flutter_gallery__transition_perf_e2e_ios to be unflaky (#110185)

newtaDev committed:
use super parameters in framework (#103620)

fluttergithubbot committed:
Marks Mac_ios flutter_gallery_ios_sksl_warmup__transition_perf to be unflaky (#110187)

fluttergithubbot committed:
Marks Mac_ios flutter_gallery_ios__transition_perf to be unflaky (#110186)

fluttergithubbot committed:
Marks Windows complex_layout_win_desktop__start_up to be unflaky (#110192)

fluttergithubbot committed:
Marks Windows flutter_gallery_win_desktop__start_up to be unflaky (#110190)

fluttergithubbot committed:
Marks Windows flutter_gallery_win_desktop__compile to be unflaky (#110189)

a-wallen committed:
[macOS] Bring up "hello_world" devicelab, compilation test for x86. (#109891)

fluttergithubbot committed:
Marks Windows hello_world_win_desktop__compile to be unflaky (#110188)

fluttergithubbot committed:
Marks Windows complex_layout_win_desktop__compile to be unflaky (#110191)

engine-flutter-autoroll committed:
Roll Flutter Engine from 5ce72b373e1a to dc14649e1a51 (1 revision) (#110198)

ankur2136 committed:
Add spacing between text to prevent interferance

Paint bounds have an offset of 4 in each direction causing the histogram to pick up values from neighboring text which made the original test pass contrast guidelines even though it should fail.

ankur2136 committed:
Add spacing between text to prevent interferance

Paint bounds have an offset of 4 in each direction causing the histogram to pick up values from neighboring text which made the original test pass contrast guidelines even though it should fail.

ankur2136 committed:
Merge branch 'a11yAdditionTest' of https://github.com/ankur2136/flutter into a11yAdditionTest
